### PR TITLE
override ToString for matcher.

### PR DESCRIPTION
**Pull request by vwxyzh**

### CODE REVIEW

File: src/Microsoft.DocAsCode.MarkdownLite/Matchers/AnyCharInMatcher.cs [hunk 1]
```diff
@@ -27,5 +27,10 @@ public Matcher Repeat(int minOccur, int maxOccur)
         {
             return new AnyCharInRepeatMatcher(_ch, minOccur, maxOccur);
         }
+
+        public override string ToString()
+        {
+            return "[" + string.Join("", _ch) + "]";
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.MarkdownLite/Matchers/AnyCharInMatcher.cs]: is it more readable to use $"" style?

Comment (hellosnow) [inline src/Microsoft.DocAsCode.MarkdownLite/Matchers/AnyCharInMatcher.cs]: >"" [](start = 37, length = 2)

how about using string.Empty?

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.MarkdownLite/Matchers/CharMatcher.cs [hunk 2]
```diff
@@ -25,5 +25,10 @@ public Matcher Repeat(int minOccur, int maxOccur)
         {
             return new CharRepeatMatcher(_ch, minOccur, maxOccur);
         }
+
+        public override string ToString()
+        {
+            return _ch.ToString();
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.MarkdownLite/Matchers/CharMatcher.cs]: >_ch.ToString() [](start = 19, length = 14)

prepend `\` for reserved chars?

**Comment by qinezh** [comment]
:shipit:

**Comment by vicancy** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:shipit:

**Pull request merged**